src/interlocking_component/route_controller.py

```diff
@@ -190,16 +182,47 @@ def maybe_update_fahrstrasse(self, train: Train, track: Track):
                     # if it can not be set this simulation step.
 
                     # This frees the least route in the interlocking
-                    self.interlocking.free_route(route.yaramo_route)
+                    self._free_route(track)
                     # This may not be the best place (time) to do so,
                     # as the route schould be freed when the train leaves the route
-                    # and not when it is still on the last segment.
+                    # and not when it is still on it.
 
                     # This sets the route in SUMO.
                     # The Interlocking Route has the same id as the SUMO route.
                     train.route = interlocking_route.id
                     return
 
+    def _free_route(self, track: Track):
+        """This method frees the route corresponding to the given track.
+
+        :param track: One track of the active Route
+        :type track: Track
+        """
+        route = self._get_interlocking_route_for_track(track)
+
+        if route is not None:
+            # This frees the least route in the interlocking
+            self.interlocking.free_route(route.yaramo_route)
+
+    def _get_interlocking_route_for_track(self, track: Track) -> Route:
+        """This method returns the interlocking route corresponding to the given track.
+
+        :param track: The track to which the route is searched
+        :type track: Track
+        :return: The interlocking Route corresponding to the track
+        :rtype: Route
+        """
+        for route_candidate in self.interlocking.active_routes:
+            interlocking_track_candidat = route_candidate.contains_segment(
+                track.identifier.split("-re")[0]
+            )
```

Comment:
Is it possible that a track belongs to more than one route?

Reply:
It is not possible that a track belongs to more than one **active** route. I will add a comment about this, as this question has come up before.